File: src/main/resources/static/back/js/pagination.js
'use strict';

class Pagination {

	constructor(listSize = 10, rangeSize = 10) {
		this.listSize = listSize;
		this.rangeSize = rangeSize;
		this.page;
		this.range;
		this.listCnt;
		this.pageCnt;
		this.startPage;
		this.startList;
		this.endPage;
		this.prev;
		this.next;
		this.info = {};		
	}

	setInfo(page = 1, range = 1, listCnt = 0) {
		this.page = page;
		this.range = range;
		this.listCnt = listCnt;

		// 전체 페이지수
		this.pageCnt = Math.ceil(this.listCnt / this.listSize);

		// 시작 페이지
		this.startPage = page == 1 ? 1 : (this.range - 1) * this.rangeSize + 1;

		// 끝 페이지
		this.endPage = this.range * this.rangeSize + 1;

		// 게시판 시작번호
		this.startList = this.page == 1 ? 1 : (this.page - 1) * this.listSize + 1;

		// 이전 버튼 상태
		this.prev = this.range == 1 ? false : true;

		// 다음 버튼 상태
		this.next = this.endPage > this.pageCnt ? false : true;

		if (this.endPage > this.pageCnt) {
			this.endPage = this.pageCnt;

			this.next = false;
		}

		this.info.listSize = this.listSize;
		this.info.rangeSize = this.rangeSize;
		this.info.page = this.page;
		this.info.range = this.range;
		this.info.pageCnt = this.pageCnt;
		this.info.startPage = this.startPage;
		this.info.startList = this.startList;
		this.info.endPage = this.endPage;
		this.info.prev = this.prev;
		this.info.next = this.next;
	}

}

const setpagination = () => {
	const info = JSON.parse($('input[name=info]').val());
	const pageWrap = $('#pageWrap');
	const url = $('input[name=url]').val();
	const page = parseInt($('input[name=page]').val());
	const range = parseInt(info.range);
	const rangeSize = parseInt(info.rangeSize);
	const listCnt = parseInt(info.listCnt);
	const pagination = new Pagination();
	let prevBtn = `<li class="page-item disabled"><a class="page-link" href="#">Prev</a></li>`;
	let nextBtn = `<li class="page-item disabled"><a class="page-link" href="#">Next</a></li>`;
	let html = ``;

	info.prev = Boolean(info.prev);
	info.next = Boolean(info.next);

	// 페이징 range설정
//	pagination.rangeSize = 5;

	if (info.prev === true) {
		pagination.setInfo(page - rangeSize, range - 1, listCnt);
		prevBtn = `<li class="page-item"><a class="page-link" href="${url}?page=${pagination.startPage}&range=${pagination.info.range}">Prev</a></li>`;
	}

	html += prevBtn;
	
	for (let i = parseInt(info.startPage);i <= parseInt(info.endPage);i++)
		html += parseInt(info.page) === i ? `<li class="page-item active"><a class="page-link" href="${url}?page=${i}&range=${range}">${i}</a></li>` : `<li class="page-item"><a class="page-link" href="${url}?page=${i}&range=${range}">${i}</a></li>`;

	if (info.next === true) {
		pagination.setInfo(page + rangeSize, range + 1, listCnt);
		nextBtn = `<li class="page-item"><a class="page-link" href="${url}?page=${pagination.startPage}&range=${pagination.info.range}">Next</a></li>`;
	}

	html += nextBtn;

	pageWrap.html(html);
};

setpagination();

File: src/main/java/kr/co/doogle/pagination/Pagination.java
package kr.co.doogle.pagination;

import org.json.JSONObject;

public class Pagination {

	private int listSize = 10; // 초기값으로 목록개수를 10으로 셋팅
	private int rangeSize = 10; // 초기값으로 페이지범위를 10으로 셋팅
	private int page;
	private int range;
	private int listCnt;
	private int pageCnt;
	private int startPage;
	private int startList;
	private int endPage;
	private String prev;
	private String next;
	private JSONObject info = new JSONObject();

	public int getListSize() {
		return listSize;
	}

	public void setListSize(int listSize) {
		this.listSize = listSize;
	}

	public int getRangeSize() {
		return rangeSize;
	}

	public void setRangeSize(int rangeSize) {
		this.rangeSize = rangeSize;
	}

	public int getPage() {
		return page;
	}

	public void setPage(int page) {
		this.page = page;
	}

	public int getRange() {
		return range;
	}

	public void setRange(int range) {
		this.range = range;
	}

	public int getListCnt() {
		return listCnt;
	}

	public void setListCnt(int listCnt) {
		this.listCnt = listCnt;
	}

	public int getPageCnt() {
		return pageCnt;
	}

	public void setPageCnt(int pageCnt) {
		this.pageCnt = pageCnt;
	}

	public int getStartPage() {
		return startPage;
	}

	public void setStartPage(int startPage) {
		this.startPage = startPage;
	}

	public int getStartList() {
		return startList;
	}

	public void setStartList(int startList) {
		this.startList = startList;
	}

	public int getEndPage() {
		return endPage;
	}

	public void setEndPage(int endPage) {
		this.endPage = endPage;
	}

	public String isPrev() {
		return prev;
	}

	public void setPrev(String prev) {
		this.prev = prev;
	}

	public String isNext() {
		return next;
	}

	public void setNext(String next) {
		this.next = next;
	}

	public void setInfo(JSONObject info) {
		this.info = info;
	}

	public void setInfo(int page, int range, int listCnt) {
		this.page = page;
		this.range = range;
		this.listCnt = listCnt;

		// 전체 페이지수
		pageCnt = (int) Math.ceil(listCnt / listSize);

		// 시작 페이지
		startPage = page == 1 ? 1 : (range - 1) * rangeSize + 1;

		// 끝 페이지
		endPage = range * rangeSize;

		// 게시판 시작번호
		startList = page == 1 ? 1 : (page - 1) * listSize + 1;

		// 이전 버튼 상태
		prev = range == 1 ? "" : "true";

		// 다음 버튼 상태
		next = endPage > pageCnt ? "" : "true";

		if (endPage > pageCnt) {
			endPage = pageCnt;

			next = "";
		}
	}

	public String getInfo() {
		info.put("listCnt", Integer.toString(listCnt));
		info.put("listSize", Integer.toString(listSize));
		info.put("rangeSize", Integer.toString(rangeSize));
		info.put("page", Integer.toString(page));
		info.put("range", Integer.toString(range));
		info.put("pageCnt", Integer.toString(pageCnt));
		info.put("startPage", Integer.toString(startPage));
		info.put("startList", Integer.toString(startList));
		info.put("endPage", Integer.toString(endPage));
		info.put("prev", String.valueOf(prev));
		info.put("next", String.valueOf(next));

		return info.toString();
	}

}